Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/child_app/view/MainActivity.java

```diff
@@ -20,23 +20,20 @@
 
     ChildPlan activePlan;
 
-    TextView planNameWidget;
-
     @Override
     protected void onCreate(Bundle savedInstanceState) {
+        getIntent().addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
         super.onCreate(savedInstanceState);
         ((App) getApplication()).getAppComponent().inject(this);
         activePlan = childPlanRepository.getActivePlan();
         if (activePlan != null) {
-            Intent planList = new Intent(MainActivity.this, TaskListActivity.class);
-            startActivity(planList);
+            Intent taskList = new Intent(MainActivity.this, TaskListActivity.class);
+            startActivity(taskList);
             setUpView();
         }
         //Todo - else: redirect to finish activity
     }
 
-    private void setUpView() {
-        setContentView(R.layout.child_app_activity_main);
-    }
+    private void setUpView() { }

```

Comment:
If it should be removed, it should method setUpView and xml view should be removed as well.
Not sure if should though. I would rahter move setUpView method under if statement. Original app has default view with happy face if there is no plan / plan was done. That's how that view was supposed to work.

Reply:
true, done